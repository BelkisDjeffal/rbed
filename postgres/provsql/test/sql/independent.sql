\set ECHO none
SET search_path TO provsql_test,provsql;

-- Will fail because this is not independent
CREATE TABLE independent_result AS
SELECT city, probability_evaluate(provenance(),'independent') AS prob
FROM (
  SELECT DISTINCT city
  FROM personnel
EXCEPT 
  SELECT p1.city
  FROM personnel p1,personnel p2
  WHERE p1.id<p2.id AND p1.city=p2.city
  GROUP BY p1.city
) t
ORDER BY CITY;

CREATE TABLE independent_result AS
SELECT city, probability_evaluate(provenance(),'independent') AS prob
FROM personnel
GROUP BY city
ORDER BY city;

SELECT remove_provenance('independent_result');

SELECT city, ROUND(prob::numeric,2) AS prob FROM independent_result;
DROP TABLE independent_result;
