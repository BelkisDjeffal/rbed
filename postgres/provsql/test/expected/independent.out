\set ECHO none
ERROR:  Not an independent circuit
 remove_provenance 
-------------------
 
(1 row)

   city   | prob 
----------+------
 Berlin   | 0.82
 New York | 0.28
 Paris    | 0.86
(3 rows)

